cmd/snap-confine/mount-support.c

```diff
@@ -641,8 +641,9 @@ static bool is_mounted_with_shared_option(const char *dir)
 void sc_ensure_shared_snap_mount()
 {
 	if (!is_mounted_with_shared_option("/")
-	    && !is_mounted_with_shared_option("/snap")) {
-		sc_do_mount("/snap", "/snap", "none", MS_BIND | MS_REC, 0);
+	    && !is_mounted_with_shared_option(SNAP_MOUNT_DIR)) {
+		sc_do_mount(SNAP_MOUNT_DIR, "/snap", "none", MS_BIND | MS_REC,
```

Comment:
As discussed on IRC this needs to say `SNAP_MOUNT_DIR` instead of `"/snap"` in the second argument. If you can please push this with a more comprehensive commit message as well.